atlp-rwanda/e-commerce-lydia-32-fn
# Ft buyer should be able to view their cart#187570835

### What does this PR do?
This PR ensures that users are able to view their cart, including product details such as product name, category, image, price, and the quantity of each item. It also allows users to update the quantity of items in their cart and delete items from their cart.
### Description of Task to be completed?

1. Fetch cart items from the API and display them on the cart page.
2. Map cart items to corresponding product details.
3. Display product name, category, image, price, and quantity.
4. Allow users to update the quantity of items in their cart.
5. Allow users to delete items from their cart.
6. Ensure the UI is responsive and displays correctly on different screen sizes.
7.Handle loading and error states for fetching cart items

### How should this be manually tested?
1. Ensure the backend server is running and accessible.
2. Login and Navigate to the cart page in the application.
3. Verify that the cart items are displayed with all relevant details.
4. Update the quantity of an item and ensure the changes are reflected correctly.
5. Delete an item from the cart and verify that it is removed from the list.

### What are the relevant pivotal tracker stories
https://www.pivotaltracker.com/story/show/187570835
